common/wallet_tx.c

```diff
@@ -39,6 +48,59 @@ struct command_result *param_wtx(struct command *cmd,
         return NULL;
 }
 
+struct command_result *param_wtx_utxos(struct command *cmd,
+				 const char *name,
+				 const char *buffer,
+				 const jsmntok_t *tok,
+				 struct wallet_tx *wtx)
+{
+	size_t i;
+	const jsmntok_t *curr;
+	struct bitcoin_txid **txids;
+	unsigned int **outnums;
+	char **outpoint = tal_arr(tmpctx, char*, 0);
+
+	txids = tal_arr(cmd, struct bitcoin_txid*, 0);
+	outnums = tal_arr(cmd, unsigned int*, 0);
+	json_for_each_arr(i, curr, tok) {
+		outpoint = tal_strsplit(cmd, json_strdup(cmd, buffer, curr), ":",
+				STR_NO_EMPTY);
+		if (tal_count(outpoint) < 3 || strlen(outpoint[0]) != 64)
```

Comment:
Why are you expecting 'outpoint' to be at least 3 here? Why not check that it's exactly 3? Moving 3 and 64 to constants might be a better way of expressing what you're checking as well.

This would be easier to review if you had included an example of the expected rpc call in the commit notes.

Reply:
I test the size of the outpoint array (which always ends with NULL and must have 2elements). I put this condition while testing and should have changed it.

Reply:
Ok cool. Moving to JSON format like rusty recommended above will probably change this too :)

Reply:
I don't think, since I need to decode the txid and the outpoint out of the jmn token

Reply:
Ok, finally made a json_helper for json token to transaction id :D

Reply:
Oh, I just saw that you did the same in your external wallet funding https://github.com/ElementsProject/lightning/pull/2672/commits/d8dcbce4485b91a9b4e6f9037270fedf18b19002